### PR TITLE
support gemmK padding

### PR DESCRIPTION
**Pull request by asleepzzz**
this PR support gemmK padding,
gemmM and gemmN is on going

### CODE REVIEW

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: `is_filter_pad` -> `isFilterPad` to align variable naming convention

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: to align variable naming convention, please remove all `_`

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: should be `gemmDim0Name` to align with variable naming convention.

File: mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir
Comment (whchung) [inline mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir:7]: the dimension of these memrefs should not be changed

File: mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir
Comment (whchung) [inline mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir:51]: the dimension of these memrefs should not be changed

**Review by whchung — DISMISSED** [review]
1) variable names should be changed to follow the naming convention.
2) memref dimensions should not be changed in the function signature as memref allocation is handled in the host code, NOT by `miopen.conv2d`.

**Comment by whchung** [comment]
Will hold off merging this PR until issues found in #195 which is caused by #180 been addressed.

File: mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir
Comment (asleepzzz) [reply mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir:7]: gemmM, gemmN padding not support yet ,turn off it until function complete

File: mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir
Comment (asleepzzz) [reply mlir/test/Dialect/MIOpen/lowering_padding_kernel.mlir:51]: gemmM, gemmN padding not support yet ,turn off it until function complete

**Comment by asleepzzz** [comment]
this PR may not add ci  and may have conflicts until right pad PR merged

**Comment by whchung** [comment]
Depend on #202 gets reviewed / merged first.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: `isInputPad`

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: `isOutputPad`

**Review by whchung — CHANGES_REQUESTED** [review]
some variables are changed to remove `_` but not all of them.

**Comment by asleepzzz** [comment]
jenkins: retest

**Comment by whchung** [comment]
jenkins: retest this plase.

@asleepzzz be polite with the bot otherwise it won't function.

**Comment by whchung** [comment]
jenkins: retest this please.

@asleepzzz be polite with the bot otherwise it won't function.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: variables names need to be changed.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: variable names need to be changed.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: variable names need to be changed

**Review by whchung — CHANGES_REQUESTED** [review]
additional variable names need to be changed.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: comment is needed to explain the general idea behind the algorithm.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h:1205]: same as above, comment is needed to explain the general idea behind the algorithm.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: same as above, comment is needed to explain the general idea behind the algorithm.

also, matrix C is only padded in case of backward weight right now, not forward convolution. it needs to be documented.

File: mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h
Comment (whchung) [inline mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h]: `gemmCPad`

File: mlir/test/mlir-miopen-driver/e2e/padding_kernel_gemmK.mlir
Comment (whchung) [inline mlir/test/mlir-miopen-driver/e2e/padding_kernel_gemmK.mlir:1]: please add the following combinations:
- `-rand 1`
- `-x2`
- `-t f16`

also please work with @yiqian1 to enable these configs in nightly CI runs.

**Comment by whchung** [comment]
@yiqian1 please work with @asleepzzz to add those `C=3` cases into nightly CI runs after this PR gets in.

The PR is in good shape and with a couple of variable naming changes and comments to explain the algorithm it would be merged.

File: mlir/test/mlir-miopen-driver/e2e/padding_kernel_gemmK.mlir
Comment (asleepzzz) [reply mlir/test/mlir-miopen-driver/e2e/padding_kernel_gemmK.mlir:1]: is default host already gfx908?

Comment (whchung) [reply mlir/test/mlir-miopen-driver/e2e/padding_kernel_gemmK.mlir:1]: NO. We have 2 CI nodes, one on gfx900, the other on gfx908. CHECK with @yiqian1 on how to automatically set `-x2`.

**Review by whchung — APPROVED** [review]

**Pull request merged**
